lib/plugin/css-variables-collector.js

```diff
@@ -27,27 +28,41 @@ CSSVariablesCollectorPlugin.prototype = {
 		return this.ruleStack.length > 0 && !this.ruleStack[this.ruleStack.length - 1].variable;
 	},
 
-	_isInGlobalOrBaseImport() {
-		return this.config.libName !== "sap.ui.core" && this.importStack.filter((importNode) => {
-			return /\/(?:global|base)\.less$/.test(importNode.importedFilename);
-		}).length > 0;
+	_isVarInLibrary({varName, filename}) {
+		// for libraries we check that the file is within the libraries theme path
+		// in all other cases with no filename (indicates calculated variables)
+		// or in case of variables in standalone less files we just include them!
+		const include = !filename ||
+			(this.config.libPath ? filename.startsWith(`/resources/${this.config.libPath}/themes/`) : true);
```

Comment:
We can't rely on the fact that the path starts with `/resources/`. In the grunt/connect use-case they just start with `/sap/...`.
The current solution in https://github.com/SAP/less-openui5/pull/168 is to lookup the library namespace within the path. Not a bullet-proof solution in case an absolute real FS path is used, but it will at least support the both main use cases (UI5 Tooling + grunt/connect).

Reply:
Ok, so more my initial regex approach again!